File: funeral-company-frontend/src/app/contract.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class ContractService {

  uri = 'http://localhost:8080'

  constructor(private http: HttpClient) { }

  getAllContracts(){
    return this.http.get(`${this.uri}/contracts`)
  }

  getAllValidContracts(){
    return this.http.get(`${this.uri}/contracts/valid`)
  }

  getAllContractsByClient(clientID){
    return this.http.get(`${this.uri}/contracts/client/${clientID}`)
  }

  getSumOfAllContractsByClient(clientID){
    return this.http.get(`${this.uri}/contracts/client/sum/${clientID}`)
  }  

  getNewestContractByClient(clientID){
    return this.http.get(`${this.uri}/contract/client/newest/${clientID}`)
  }  

  saveContract(clientID, dateSigned, paymentDate, validUntil, money, currency){
    const data = {
      dateSigned: dateSigned,
      paymentDate: paymentDate,
      validUntil: validUntil,
      money: money,
      currency: currency,
      client: clientID,
      valid: true
    }

    return this.http.post(`${this.uri}/contract`, data)
  }

  changeValid(contractID){
    const data = {
      
    }
    return this.http.post(`${this.uri}/contract/valid/${contractID}`, data)
  }

  removeContract(contractID){    
    return this.http.delete(`${this.uri}/contract/${contractID}`)
  }
}

File: funeral-company-backend/src/main/java/rs/ac/bg/etf/funeral/company/backend/service/ContractService.java
package rs.ac.bg.etf.funeral.company.backend.service;

import rs.ac.bg.etf.funeral.company.backend.entity.Client;
import rs.ac.bg.etf.funeral.company.backend.entity.Contract;

import java.util.List;

public interface ContractService {

    List<Contract> getAllContracts();
    List<Contract> getAllValidContracts();
    Contract saveContract(Contract contract);
    Contract updateValidContract(Long contractID);
    List<Contract> getAllContractsByClient(Long clientID);
    List<Object[]> sumMoneyOfClient(Long clientID);
}
